plugins/dicom_viewer/server/__init__.py

```diff
@@ -195,9 +197,46 @@ def _uploadHandler(event):
     events.trigger('dicom_viewer.upload.success')
 
 
+def dicomSubstringSearchHandler(query, types, user=None, level=None, limit=0, offset=0):
+    """
+    Provide a substring search on both keys and values.
+    """
+    if types != ['item']:
+        raise RestException('The dicom search is only able to search in Item.')
+
+    # The insensitive Case is not perfect
+    # We will need to search into object value as improvment.
+    jsQuery = """
+        function() {
+            const key = '%(key)s'.toLowerCase();
+            const value = '%(value)s'.toLowerCase();
+            const meta = obj.dicom.meta;
+            return Object.keys(meta).some(
+                function(key) {
+                    return (key.toLowerCase().indexOf(key) !== -1)  ||
+                        meta[key].toString().toLowerCase().indexOf(value) !== -1;
+                })
+            }
+        """ % {'key': query, 'value': query}
+
+    # Sort the documents inside MongoDB
+    cursor = Item().find({'dicom': {'$exists': True}, '$where': jsQuery})
+    # Filter the result
+    result = Item().filterResultsByPermission(cursor, user, level, limit, offset)
```

Comment:
You can inline this `filterResultsByPermission` into the list comprehension below.

Reply:
What do you mean by inline this ?

Reply:
Combine this line and the next.

Reply:
But `filterResultsByPermission` take a cursor, and `filter` take only one element, how can I compatc them into one line ?
Something like: 
```
result = {
        'item': Item().filterResultsByPermission([Item().filter(d, user) for d in result], user, level ...)
}
```
is ok ?

Reply:
`'item': [Item().filter(i, user) for i in Item().filterResultsByPermission(...)]`

Reply:
mmh yes it make sens! Thanks!

Reply:
In order to fit in the 100 characters limitation what is the best : 
```
result = {
        'item': [Item().filter(d, user) 
            for d in Item().filterResultsByPermission(cursor, user, level, limit, offset)]
    }
``` 
or 
```
    result = {
        'item': [Item().filter(d, user) for d in Item().filterResultsByPermission(
            cursor, user, level, limit, offset)]
    }
```
or again
```
    result = {
        'item': [
            Item().filter(d, user)
            for d in Item().filterResultsByPermission(cursor, user, level, limit, offset)
        ]
    }

```

Reply:
I personally like the third one, but the second one is OK too.

Reply:
Me too, but is this easily readable ?

Reply:
I think it is, it's a fairly common way to break up a list comprehension. If you don't think it's readable enough, you could always make a variable called `cursor` and use that in the comprehension instead of inlining the `filterResultsByPermission` call.

Reply:
That's good to me, thank you Zach!